docs/source/onedal/algorithms/clustering/kmeans.rst

```diff
@@ -85,10 +85,16 @@ to each cluster.
 .. math::
    c_j^{(t + 1)} = \frac{1}{|S_j^{(t)}|} \sum_{x \in S_j^{(t)}} x, \quad 1 \leq j \leq k.
 
+If it turned out that some of :math:`S_j^{(t)}` are empty then start empty clusters handling procedure.
+In this case set :math:`c_j^{(t + 1)}` equal to the farthest point from just calculated centroids for each empty cluster.
```

Comment:
Hmm maybe something like:

In this case, you can set the value of :math:c_j^{(t+1)} as the farthest point from the previously calculated centroids for each empty cluster. It fills in the empty clusters with new data points and makes sure that all clusters have a representative centroid.

Reply:
I'm not really sure that it makes sense to write 'fill' here because we are not filling the new clusters at this step. 
If you want to highlight the purpose of the procedure then what about this:

In this case, you can set the value of :math:c_j^{(t+1)} as the farthest point from the previously calculated centroids for each empty cluster. This procedure makes sure that the number of cluster remains the same.

Reply:
I like your suggestion!

Reply:
done